app/email/email.go

```diff
@@ -604,6 +604,147 @@ func (es *Service) SendGuestInviteEmails(team *model.Team, channels []*model.Cha
 	return nil
 }
 
+func (es *Service) SendInviteEmailsToTeamAndChannels(
+	team *model.Team,
+	channels []*model.Channel,
+	senderName string,
+	senderUserId string,
+	senderProfileImage []byte,
+	invites []string,
+	siteURL string,
+	reminderData *model.TeamInviteReminderData,
+	message string,
+	errorWhenNotSent bool,
+) error {
+	if es.perHourEmailRateLimiter == nil {
+		return NoRateLimiterError
+	}
+	rateLimited, result, err := es.perHourEmailRateLimiter.RateLimit(senderUserId, len(invites))
+	if err != nil {
+		return SetupRateLimiterError
+	}
+
+	if rateLimited {
+		mlog.Error("rate limit exceeded", mlog.Duration("RetryAfter", result.RetryAfter), mlog.Duration("ResetAfter", result.ResetAfter), mlog.String("user_id", senderUserId),
+			mlog.String("team_id", team.Id), mlog.String("retry_after_secs", fmt.Sprintf("%f", result.RetryAfter.Seconds())), mlog.String("reset_after_secs", fmt.Sprintf("%f", result.ResetAfter.Seconds())))
+		return RateLimitExceededError
+	}
+
+	channelsLen := len(channels)
+
+	subject := i18n.T("api.templates.invite_team_and_channels_subject_a", map[string]interface{}{
```

Comment:
we could use the keys `invite_team_and_channels_subject` and `invite_team_and_channel_subject` (no s at channel, or change with `single_channel`) to avoid the a/b suffixes and have more explicit names

Reply:
make sense, thanks!

Reply:
nice catch! solved